### DISCUSSION TITLE
make linux64 error

### DISCUSSION THREAD

**Discussion by astroJarHead** [iraf-community/iraf#245] [Q&A] [answered]
The install script completed successfully. I receive an error with make linux64:
```
iraf]: make linux64
(util/mkarch linux64)
Archs='linux64'
Making architecture: linux64
hlib: Undefined variable.
envinit: cannot open `/mnt/nofs/opt/nofs/iraf/noao/noaolib/zzsetenv.def'
Warning, mkpkg line 0: cannot open `noao$lib/mkpkg.inc'
hlib: Undefined variable.
make: *** [linux64] Error 1
```
The script mkarch is util/running based on the output I see. I do not see a directory `noao/noaolib/` but there is a directory `noao/lib/` and this contains a file `zzsetenv.def` in there. 

Should a directory `noao/noaolib/` have been created during an earlier part of the installation? 

The `noao/lib/` directory does contain a `mkpkg.inc` file. Perhaps there is a difficulty expanding `noao$` to the path  leading to the `lib/` directory?

**Comment by olebole** [comment, marked as answer]
Since this is most probably not a bug, I moves this to the discussion forum.

The question here is whether you have remnants (environment variables or setup files) from an old IRAF installation. Is `/mnt/nofs/opt/nofs/iraf/` the path of the IRAF installation you are going to create? 
Please check in the file `~/.iraf/setup.sh` you find a line `export iraf=…` with the correct path to your installation is set, and that the `iraf` environment variable is *not* set (or set to the current path) before running `make linux64`.

Specifically the line `envinit: cannot open …`  is very suspicious, since this is the output of a compiled program and not a shell script; however when doing `make linux64` nothing is compiled yet.

**Reply by astroJarHead** [reply]
Thank you for the prompt reply. 

From `~/.iraf/setup.sh` I see: `export iraf=/opt/nofs/iraf/`

and `$iraf` is not set:
```
nofs]: printenv $iraf
iraf: Undefined variable.
```
admittedly without the preceding `/mnt/nofs/` . I did edit the `~/.iraf/setup.sh` to test whether adding the `/mnt/nofs/` would help. I did not think so given our system design. The output was the same:
```
iraf]: make linux64
(util/mkarch linux64)
Archs='linux64'
Making architecture: linux64
hlib: Undefined variable.
envinit: cannot open `/mnt/nofs/opt/nofs/iraf/noao/noaolib/zzsetenv.def'
Warning, mkpkg line 0: cannot open `noao$lib/mkpkg.inc'
hlib: Undefined variable.
make: *** [linux64] Error 1
```
Our IT staff did create the build as a service for me, which I was not involved in making. The build.log contains compilation warnings but no glaringly obvious signs of a problem, but honestly I am no expert in interpreting the success of the IRAF build. 
As you referred to a previous installation perhaps this is what you could mean, unless I am confusing the situation.

**Reply by olebole** [reply]
When your IT staff already successfully created the build, why do you then try to do it again? You can then just set the `$iraf` and `$IRAFARCH` variables (done in `~/.iraf/setup.py`) and start working. 
Did they actually run the test suite? In this case, this is the easiest way.

`printenv $iraf` is wrong, since the `$` already resolves the environvent variable, and `printenv` then tries to interpret the content of `$iraf` as environment variable.

**Reply by astroJarHead** [reply]
They reported the test had failed, but did not have output to go along with that and in the time that went by between their report and my attempt to run IRAF we thought to just try again. My calling "cl" ended with a segmentation fault, which led me to query the IT staff as to what they had experienced. 

Sorry for the poor attempt on the environment variable. I get an empty response back:
```
iraf]: printenv iraf
iraf]: 
```

**Reply by olebole** [reply]
In that case I would suggest to start from a clean tarball and attach the full make log here if the tests fail again.

**Reply by astroJarHead** [reply]
Thank you, I will go that route.
